sdk/state/open.go

```diff
@@ -100,10 +109,32 @@ func (c *Channel) ProposeOpen(p OpenParams) (Open, error) {
 //
 // If after confirming the open has all the signatures it needs to be fully and
 // completely signed, fully signed will be true, otherwise it will be false.
-func (c *Channel) ConfirmOpen(m Open) (open Open, authorized bool, err error) {
+func (c *Channel) ConfirmOpen(m OpenAgreement) (open OpenAgreement, authorized bool, err error) {
+	// at the end of this method, if no error, then save a new channel openAgreement. Use the
+	// channel's saved open agreement details if present, to prevent other party from changing.
+	defer func() {
+		if err != nil {
+			return
+		}
+		c.openAgreement = OpenAgreement{
+			CloseSignatures:       appendNewSignatures(c.openAgreement.CloseSignatures, m.CloseSignatures),
+			DeclarationSignatures: appendNewSignatures(c.openAgreement.DeclarationSignatures, m.DeclarationSignatures),
+			FormationSignatures:   appendNewSignatures(c.openAgreement.FormationSignatures, m.FormationSignatures),
+			Details: OpenAgreementDetails{
+				Asset:      m.Details.Asset,
+				AssetLimit: m.Details.AssetLimit,
+			},
+		}
+	}()
+
+	// validate openAgreement details
+	if !c.openAgreement.isEmpty() && m.Details != c.openAgreement.Details {
+		return m, authorized, fmt.Errorf("input open agreement details do not match the saved open agreement details")
+	}
```

Comment:
💡 The comment to validate isn't very specific, how about:
```suggestion
	// If the open agreement details don't match the open agreement in progress, error.
	if !c.openAgreement.isEmpty() && m.Details != c.openAgreement.Details {
		return m, authorized, fmt.Errorf("open agreement details do not match the open agreement in progress")
	}
```

Reply:
❗ Could you move this check above the defer? I think it's a little more defensive to place code that protects the rest of the function ahead of any code, and we don't need the defer to run if this errors.

Reply:
sounds good, I moved the update validation step above the defer too to keep the same pattern
https://github.com/stellar/experimental-payment-channels/pull/125/commits/9a1e0b0da100dd44f31815753c5d143c1f947907